nethttp/client.go

```diff
@@ -193,7 +193,7 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 		carrier := opentracing.HTTPHeadersCarrier(req.Header)
 		tracer.sp.Tracer().Inject(tracer.sp.Context(), opentracing.HTTPHeaders, carrier)
 	}
-
+	req = req.WithContext(opentracing.ContextWithSpan(req.Context(), tracer.sp))
```

Comment:
the span (`tracer.sp`) is created and returned in L185. I would suggest doing this there:
```
        sp := tracer.start(req)
        req = req.WithContext(opentracing.ContextWithSpan(req.Context(), sp))
```